lib/msf/ui/console/driver.rb

```diff
@@ -589,6 +591,15 @@ def on_variable_set(glob, var, val)
     end
   end
 
+
+  def is_payload_compatible?(m, payload_name)
```

Comment:
Yeah sure, it would be pretty much like the compatible_payloads method. But instead of return an array, it can just return true as soon as there's a compatibility match.